packages/components/src/AppGuidedTour/AppGuidedTour.component.js

```diff
@@ -73,8 +75,10 @@ function AppGuidedTour({
 			onRequestClose={() => {
 				onRequestClose();
 				setIsAlreadyViewed(true);
-				setCurrentStep(0);
-				setImportDemoContent(false);
+				if (importDemoContent) {
+					setImportDemoContent(false);
+					setCurrentStep(currentStep - 1);
```

Comment:
Can `currentStep `be equal to 0 or less here ?
```suggestion
					setCurrentStep(Math.max(0, currentStep - 1));
```

Reply:
@aamalric-talend 
Thank you for your comment, even pass  empty `steps`, it always inserts a welcome page, so I think the `currentStep` should not be less 0, but I prefer your changes, it is more safe, thank you!